File: DATA_STRUCTURE/MORNINGSESSION/ARRAY/goodPair.java

// FIND GOOD PAIR IN GIVEN ARRAY

class Solution{
	public int goodPair(int arr[],int n,int key){
		for(int i=0 ; i<n ; i++){
			for(int j=i+1 ; j<n ; j++){
				if(arr[i] + arr[j] == key){
					return 1;
				}
			}
		}
		return 0;
	}
	public static void main(String[]args){
		Solution s = new Solution();
		int[] arr = {1,2,2};
		int key = 4;
		System.out.println("RESULT : "+s.goodPair(arr,3,key));
	}
}

File: DATA_STRUCTURE/ASSIGNMENT/RECURSION/goodPair.c
#include <stdio.h>

int goodPair(int arr[],int n,int k,int i,int j){

	if(i>n && j>n){
		return 0;
	}

	if(j>n){
		i=i+1;
		j=-1;
	}

	if( (i != j && j>=0 && i>j) && (arr[i]+arr[j] == k)){
		return 1 + goodPair(arr,n,k,i,j+1);
	}else{
		return goodPair(arr,n,k,i,j+1);
	}
}
void main(){
	int arr[] = {1,2,3,4};
	int n = 4 ;
	int k = 4;
	int i=0,j=0;
	int ret = goodPair(arr,n-1,k,i,j);
	printf("Count : %d\n",ret);
}
